File: Daily/sbgo/Leetcode1423-可获得的最大点数.go
package sbgo

func maxScore(cardPoints []int, k int) int {
	n := len(cardPoints)
	windowSize := n - k
	sum := 0
	for _, val := range cardPoints[:windowSize] {
		sum += val
	}
	minSum := sum
	for i := windowSize; i < n; i++ {
		sum += (cardPoints[i] - cardPoints[i-windowSize])
		minSum = min(minSum, sum)
	}
	accu := 0
	for _, val := range cardPoints {
		accu += val
	}
	ans := accu - minSum
	return ans
}

File: Daily/PY/Leetcode1423-可获得的最大点数.py
from typing import List
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)
        windowsize = n - k
        s = sum(cardPoints[:windowsize])
        minsum = s
        for i in range(windowsize,n):
            s += (cardPoints[i] - cardPoints[i - windowsize])
            minsum = min(minsum,s)
        return sum(cardPoints) - minsum
        